dev/bots/test.dart

```diff
@@ -260,6 +262,30 @@ Future<void> _runToolTests() async {
       .map<String>((String name) => path.basenameWithoutExtension(name)),
     // The `dynamic` on the next line is because Map.fromIterable isn't generic.
     value: (dynamic subshard) => () async {
+      if (subshard == 'commands') {
```

Comment:
Do the subdirectories of test/general.shard/ need this treatment? If so, I think a more manual walk of the test directory tree would be preferable here.

Reply:
Surprisingly, it seems like pub is running these tests as expected

Reply:
On further thought, I'm wondering about the wisdom of trusting it to do that before https://github.com/dart-lang/test/issues/1116 is resolved.

Reply:
Agreed, but I still think this change is an improvement overall.

I also experimented with just passing every test file individually, but that hit the terminal line limit on windows and still seemed to be skipping tests.